src/plugins/discover/public/embeddable/components/search_embeddable_field_stats_table_component.tsx

```diff
@@ -37,8 +40,30 @@ export function SearchEmbeddablFieldStatsTableComponent({
     api.fetchContext$,
     api.savedSearch$
   );
-
+  const { euiTheme } = useEuiTheme();
   const isEsql = useMemo(() => isEsqlMode(savedSearch), [savedSearch]);
+  if (isEsql) {
+    return (
+      <EuiEmptyPrompt
+        icon={<EuiIcon size="l" type="warning" color="warning" />}
+        color="plain"
+        paddingSize="m"
+        css={css`
+          margin: ${euiTheme.size.xl} auto;
+        `}
+        title={
+          <h2 data-test-subj="fieldStatsUnavailableCalloutTitle">
+            <FormattedMessage
+              id="discover.fieldStatsNotAvailableForESQLCalloutTitle"
+              defaultMessage="Field statistics are not available for ES|QL queries"
+            />
+          </h2>
+        }
+        titleSize="xs"
+        hasBorder
+      />
+    );
+  }
```

Comment:
```suggestion
```

Can we move this message from `FieldStatisticsTab` and `SearchEmbeddablFieldStatsTableComponent` to `FieldStatisticsTable` to avoid duplications? 

Also, would be great to have the same styles and copy for a search embeddable panel and a field stats panel. Currently it looks like this:
<img width="2156" alt="Screenshot 2024-11-06 at 12 35 47" src="https://github.com/user-attachments/assets/003d9716-90f1-420b-8a40-709bf9ee06d9">

Reply:
Updated here [`dc99c68` (#197538)](https://github.com/elastic/kibana/pull/197538/commits/dc99c68dd97df4e6ccae2f559d012c89ed555103)